extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildAggregatorFactory.java

```diff
@@ -114,4 +118,19 @@ public int getMaxIntermediateSize()
     return HllSketch.getMaxUpdatableSerializationBytes(getLgK(), TgtHllType.valueOf(getTgtHllType()));
   }
 
+  private void validateInputs(@Nullable ColumnCapabilities capabilities)
+  {
+    if (capabilities != null) {
+      if (capabilities.is(ValueType.COMPLEX)) {
+        throw new ISE(
+            "Invalid input type [%s] in ingestion for metric type %s, in aggregate %s for field name %s",
+            capabilities.asTypeString(),
+            HllSketchModule.BUILD_TYPE_NAME,
+            getName(),
+            getFieldName()
+        );
```

Comment:
this method is used at both query and ingestion time, and shouldn't mention ingestion
```suggestion
        throw new ISE(
            "Invalid input [%s] of type [%s] for [%s] aggregator [%s]",
            getFieldName(),
            capabilities.asTypeString(),
            HllSketchModule.BUILD_TYPE_NAME,
            getName()
        );
```